test/functional/wallet_v3_txs.py

```diff
@@ -0,0 +1,560 @@
+#!/usr/bin/env python3
+# Copyright (c) 2025 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test how the wallet deals with TRUC transactions"""
+
+from decimal import Decimal, getcontext
+
+from test_framework.authproxy import JSONRPCException
+from test_framework.messages import (
+    COIN,
+    CTransaction,
+    CTxOut,
+)
+from test_framework.script import (
+    CScript,
+    OP_RETURN
+)
+
+from test_framework.script_util import bulk_vout
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    assert_greater_than,
+    assert_raises_rpc_error,
+)
+
+from test_framework.mempool_util import (
+    TRUC_MAX_VSIZE,
+    TRUC_CHILD_MAX_VSIZE,
+)
+
+def cleanup(func):
+    def wrapper(self, *args):
+        try:
+            func(self, *args)
```

Comment:
In 09c9f828250e3840d7557374ff568204305e2679 "test: add truc wallet tests"

Since all of the test cases do a `self.generate`, that could be added here as well before `func`.